lib/rom/command.rb

```diff
@@ -84,7 +86,25 @@ def execute(*)
     #
     # @api public
     def call(*args, &block)
-      tuples = execute(*(curry_args + args), &block)
+      tuples =
+        if hooks?
+          prepared =
+            if curried?
+              apply_hooks(before_hooks, *curry_args, *args)
+            else
+              apply_hooks(before_hooks, *args)
+            end
+
+          result = prepared ? execute(prepared, &block) : execute(&block)
+
+          if curried?
+            apply_hooks(after_hooks, result, *args)
+          else
+            apply_hooks(after_hooks, result)
+          end
+        else
+          execute(*(curry_args + args), &block)
```

Comment:
yeah I know there are perf differences, I wanted to look into that closer but...you did that already, thank you :) we can tweak this in master now